u2flib_host/authenticate.py

```diff
@@ -103,17 +103,16 @@ def parse_args():
 def main():
     args = parse_args()
 
-    facet = text_type(args.facet, sys.stdin.encoding or sys.getdefaultencoding())
+    facet = text_type(args.facet)
     if args.infile:
         with open(args.infile, 'r') as f:
             data = f.read()
     else:
         if sys.stdin.isatty():
             sys.stderr.write('Enter AuthenticateRequest JSON data...\n')
-        data = sys.stdin.read()
-
-    params = json.loads(data, object_hook=u2str)
+        data = sys.stdin.readline()
```

Comment:
Switching  read() to readline() means input has to be entered on a single line. Since the input here is a JSON object, it's pretty common to have those on multiple lines. I would prefer to keep that possible.